library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -463,6 +519,9 @@ public void setup(OperatorContext context)
       currentFile = null;
       offset = 0;
     }
+
+    fileQueue = new ArrayBlockingQueue<Path>(queueCapacity);
+    scanDirService = Executors.newScheduledThreadPool(1);
```

Comment:
I followed the design in AbstractJdbcPollInputOperator. I had thought of using Executors.newSingleThreadScheduledExecutor() as we need to specify delay in the scheduling. The implementation of that one is using ScheduledThreadPoolExecutor(1). So essentially the same.